### PR TITLE
Modernize code for OTSU and reduce public API further

### PR DESCRIPTION
**Pull request by stweil**
Remove thresholder.h from the public API.

Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

**Comment by egorpugin** [comment]
Do we put it in 5.0?

**Comment by stweil** [comment]
> Do we put this into 5.0?

If nobody objects, yes.

File: include/tesseract/baseapi.h [hunk 1]
```diff
@@ -342,11 +342,7 @@ class TESS_API TessBaseAPI {
    * Note that Tesseract takes ownership of the Thresholder and will
    * delete it when it it is replaced or the API is destructed.
    */
-  void SetThresholder(ImageThresholder *thresholder) {
-    delete thresholder_;
-    thresholder_ = thresholder;
-    ClearResults();
-  }
+  void SetThresholder(ImageThresholder *thresholder);
```
Comment (egorpugin) [inline include/tesseract/baseapi.h:345]: This can't be a public API anymore, since user does not know how to inherit from ImageThresholder.

Comment (egorpugin) [reply include/tesseract/baseapi.h:345]: So should be removed in this case.

Comment (stweil) [reply include/tesseract/baseapi.h:345]: Right, so that should be removed, too, if it is not used anywhere.

Comment (stweil) [reply include/tesseract/baseapi.h]: I added a commit now which removes `TessBaseAPI::SetThresholder`, too.

**Comment by stweil** [comment]
> If nobody objects, yes.

I merged the API change now. Let's see whether someone misses the removed parts. I don't think so, but if I was wrong, we can review that decision again.

**Comment by egorpugin** [comment]
The image can be passed already thresholded.
Even preprocessed in more ways than tess could do itself.

**Comment by amitdo** [comment]
IMO, you  removed a useful feature.

Unless you intend to replace it with something better. I don't think it should be removed.

Related issue: #3083

**Comment by stweil** [comment]
I looked at that code because I want to allow better binarization and had the impression that it would not help.

**Comment by amitdo** [comment]
Okay.

**Pull request merged**